apps/rush-lib/src/logic/pnpm/PnpmProjectDependencyManifest.ts

```diff
@@ -114,7 +115,30 @@ export class PnpmProjectDependencyManifest {
     }
 
     const specifier: string = `${name}@${version}`;
-    const integrity: string = shrinkwrapEntry.resolution.integrity;
+    let integrity: string = shrinkwrapEntry.resolution.integrity;
+
+    if (!integrity) {
+      // git dependency specifiers do not have an integrity entry
+
+      // Example ('integrity' doesn't exist in 'resolution'):
+      //
+      // github.com/chfritz/node-xmlrpc/948db2fbd0260e5d56ed5ba58df0f5b6599bbe38:
+      //   dependencies:
+      //     sax: 1.2.4
+      //     xmlbuilder: 8.2.2
+      //   dev: false
+      //   engines:
+      //     node: '>=0.8'
+      //     npm: '>=1.0.0'
+      //   name: xmlrpc
+      //   resolution:
+      //     tarball: 'https://codeload.github.com/chfritz/node-xmlrpc/tar.gz/948db2fbd0260e5d56ed5ba58df0f5b6599bbe38'
+      //   version: 1.3.2
+
+      integrity =
+        `${name}@${version}:${crypto.createHash('sha256').update(JSON.stringify(shrinkwrapEntry)).digest('hex')}:`;
```

Comment:
Suggest to move `crypto.createHash('sha256').update(JSON.stringify(shrinkwrapEntry)).digest('hex')` into a local variable for easier debugging.  It's a fairly big expression for a string interpolation.